### PR TITLE
Store Q-table per episodes

### PR DESCRIPTION
**Pull request by mohsen-ghaffari1992**
@wasowski 
Please let me know if there is anything that I should change.

### CODE REVIEW

File: src/main/scala/symsim/ExactRL.scala [hunk 1]
```diff
@@ -54,16 +54,18 @@ trait ExactRL[State, ObservableState, Action, Reward, Scheduler[_]]
   /** Execute a full learning episode (until the final state of agent is
     * reached).
     */
-  def learningEpisode(fR: (VF, List[Reward]), s_t: State): Scheduler[(VF, List[Reward])] =
+  def learningEpisode(fR: (VF, List[Reward], List[VF]), s_t: State): Scheduler[(VF, List[Reward], List[VF])] =
     def p(f: VF, r_acc: Reward, s: State, a: Action): Boolean = agent.isFinal(s)
```
Comment (wasowski) [inline src/main/scala/symsim/ExactRL.scala:58]: do you have any reason to keep the List[Reward] ?  This morning I thought that we could drop it.

File: src/test/scala/symsim/ExperimentSpec.scala [hunk 2]
```diff
@@ -17,14 +17,22 @@ trait ExperimentSpec[State, ObservableState, Action]
...
-    val csvString: String = r.mkString("\n")
-    val outputFile: String = "accumulative reward.csv"
-    val writer = new PrintWriter(outputFile)
+    val (q, r, qL) = setup.runQ
+    val csvStrR: String = r.mkString("\n")
+    val outputFileR: String = "accumulative reward.csv"
+    val writerR = new PrintWriter(outputFileR)
     try {
-      writer.println(csvString)
+      writerR.println(csvStrR)
```
Comment (wasowski) [inline src/test/scala/symsim/ExperimentSpec.scala:25]: I think I asked about this before. Why do we use these complex writer things. A normal `print` does not work?

**Review by wasowski — CHANGES_REQUESTED** [review]

File: src/test/scala/symsim/ExperimentSpec.scala (hunk 2 above)
Comment (mohsen-ghaffari1992) [reply src/test/scala/symsim/ExperimentSpec.scala:25]: I am writing in a file, I did not whether there is a way to write in a file using print!

Comment (wasowski) [reply src/test/scala/symsim/ExperimentSpec.scala:25]: ah ok. Let's leave it as is then. I missed that this is is to a file.

File: src/main/scala/symsim/ExactRL.scala (hunk 1 above)
Comment (wasowski) [reply src/main/scala/symsim/ExactRL.scala:58]: Yeah. I think keeping the code that we do not use around is problematic, especially that this function already has very complex type.  We move this to evaluation instead.

**Comment by wasowski** [comment]
This looks good now. I think saving the Q-Tables to files is not needed (the saving can be done later in a pull request that does the evaluation loop, and probably just for the rewards). But I merge this now, and you can fix it in that other PR, when we move saving to the agent evaluation.

**Comment by mohsen-ghaffari1992** [comment]
Yes, I see. I will remove with the next pull request.

**Pull request merged**